lib/Connection.js

```diff
@@ -70,8 +80,8 @@ class Connection extends EventEmitter {
     const id = ++this._lastId;
     const message = JSON.stringify({id, method, params});
     debugProtocol('SEND ► ' + message);
-    this._ws.send(message);
     return new Promise((resolve, reject) => {
+      this._ws.send(message);
```

Comment:
I don't think this change affects anything. Am I missing something?

Reply:
`_ws.send` can throw an exception, but the function returns a promise. If an exception is thrown, `.catch` would not work. This way the promise is returned and rejects if `send` throws an exception.